utils/docker/Dockerfile.centos.8

```diff
@@ -12,29 +12,78 @@ LABEL maintainer="daos@daos.groups.io"
+# package must be installed and the GPG key it provides must be
+# installed.
 ARG REPO_URL
-# Need to clean this up later
-# This is largely the same as the centos.7 code, however this sets priority for
-# the new repo, and leaves the default ones enabled as there is some metadata
-# that is not available locally.
-RUN if [ -n "$REPO_URL" ]; then						                           \
-      version=$(sed -e 's/.* \([0-9\.]*\)\.[0-9]*.*/\1/' /etc/centos-release); \
-      MY_REPO="${REPO_URL}repository/centos-${version}-x86_64-group/";         \
-      MY_NAME="${MY_REPO#*//}";						                           \
-      MY_NAME="${MY_NAME//\//_}";					                           \
+ARG REPO_DISTRO=""
+RUN if [ -n "$REPO_DISTRO" ]; then                              \
+      rpm --import /etc/pki/rpm-gpg/*;                          \
+      MY_REPO="${REPO_URL}${REPO_DISTRO}/";                     \
+      MY_NAME="${MY_REPO#*//}";                                 \
+      MY_NAME="${MY_NAME//\//_}";                               \
       echo -e "[${MY_NAME}]\n\
 name=created from ${MY_REPO}\n\
 baseurl=${MY_REPO}\n\
 enabled=1\n\
-priority=1\n\
-gpgcheck=False\n" >> /etc/yum.repos.d/local-centos-group.repo;                 \
+repo_gpgcheck=0\n\
+gpgcheck=1\n" >> /etc/yum.repos.d/local-centos-group.repo;      \
+      dnf -y install --disablerepo extras --disablerepo baseos  \
+          epel-release-8-8.el8 dnf-plugins-core dnf-utils;      \
+      rpm --import /etc/pki/rpm-gpg/RPM-GPG-KEY-EPEL-8;         \
+      dnf config-manager --assumeyes --quiet --disable          \
+          baseos powertools epel epel-modular extras;           \
+      sed "s/^mirrorlist_expire=0*/mirrorlist_expire=99999999/" \
+          -i /etc/dnf/dnf.conf;                                 \
+      dnf -y upgrade epel-release;                              \
+    else                                                        \
+      dnf -y install epel-release dnf-plugins-core &&           \
+      dnf config-manager --assumeyes --set-enabled powertools;  \
+    fi;                                                         \
+    dnf clean all
+
+# Currently the appstream REPO can not be part of a repo group
+ARG REPO_APSTREAM=""
+RUN if [ -n "$REPO_APPSTREAM" ]; then                           \
+      rpm --import /etc/pki/rpm-gpg/*;                          \
+      MY_REPO="${REPO_URL}${REPO_APPSTREAM}/";                  \
+      MY_NAME="${MY_REPO#*//}";                                 \
+      MY_NAME="${MY_NAME//\//_}";                               \
+      echo -e "[${MY_NAME}]\n\
+name=created from ${MY_REPO}\n\
+baseurl=${MY_REPO}\n\
+enabled=1\n\
+repo_gpgcheck=1\n\
+gpgcheck=1\n" >> /etc/yum.repos.d/local-centos-appstream.repo;  \
+      dnf config-manager --assumeyes --quiet --disable          \
+          appstream;                                            \
+      dnf clean all;                                            \
```

Comment:
There was no `dnf` commands here that would add anything to the cache, so this is unnecessary.

Reply:
Fixed in next push